app/controllers/ticket_comments_controller.rb

```diff
@@ -2,6 +2,19 @@
 
 class TicketCommentsController < ApplicationController
   before_action :authenticate_user!
+  before_action -> { check_ticket_errors(TicketLog.action_types[:create_comment]) }, only: [:create]
+
+  private def check_ticket_errors(action_type)
+    @action_type = action_type
+    @ticket = Ticket.find(params.require(:ticket_id))
+    @acting_stakeholder = TicketStakeholder.find(params.require(:acting_stakeholder_id))
+
+    render status: :bad_request, json: { error: "You are not a stakeholder for this ticket." } unless @ticket.user_stakeholders(current_user).include?(@acting_stakeholder)
+
+    render status: :unauthorized, json: { error: "You are not allowed to perform this action." } unless @acting_stakeholder.actions_allowed.include?(@action_type)
+
+    false
```

Comment:
This is now unnecessary. `before_action` does not need any explicit return value at all.

Reply:
Done